src/Analyzers/Core/Analyzers/AbstractBuiltInCodeStyleDiagnosticAnalyzer_Core.cs

```diff
@@ -76,5 +76,15 @@ public sealed override void Initialize(AnalysisContext context)
         }
 
         protected abstract void InitializeWorker(AnalysisContext context);
+
+        protected static bool IsAnalysisLevelGreaterThanOrEquals(int minAnalysisLevel, AnalyzerOptions analyzerOptions)
```

Comment:
💭 Consider using a method with this signature:

```
public static double? GetEffectiveAnalysisLevel(this AnalyzerOptions analyzerOptions)
```

You could then use this check, which will evaluate to false if the analysis level is not set:

```
if (context.Options.GetEffectiveAnalysisLevel() >= 8)
```

Reply:
Do you foresee analyzer needing analysis for anything other than a greater than or equals check? Otherwise a bool returning `IsAnalysisLevelGreaterThanOrEquals(level)` feels much better than a nullable int returning `GetEffectiveAnalysisLevel()` method.

I am also hesitant to make this an extension method on `AnalysisOptions` as this is returning `EffectiveAnalysisLevelStyle` value, i.e. analysis level specific to CodeStyle analyzers. I'd prefer to keep this as a protected helper method defined in `AbstractBuiltInCodeStyleDiagnosticAnalyzer` to scope its usage to CodeStyle analyzers.

Reply:
Ping @sharwell